wsdl-core/src/main/java/io/ballerina/wsdl/core/OperationProcessor.java

```diff
@@ -0,0 +1,263 @@
+        this.soapVersion = soapVersion;
+    }
+
+    List<Field> generateFields() {
+        List<Field> fields = new ArrayList<>();
+        List<Field> inputFields = generateInputFields();
+        List<Field> outputFields = generateOutputFields();
+        fields.addAll(inputFields);
+        fields.addAll(outputFields);
+        return fields;
+    }
+
+    private List<Field> generateInputFields() {
+        List<String> processedOpInputNames =
+                processedOperations.stream().map(op -> op.getOperationInput().getName()).toList();
+        if (processedOpInputNames.contains(wsdlOperation.getOperationInput().getName())) {
+            return new ArrayList<>();
+        }
+        List<Field> fields = new ArrayList<>();
+        PartProcessor partProcessor = new PartProcessor();
+
+        List<WSDLPart> inputParts = new ArrayList<>();
+        List<WSDLPart> inputHeaderParts =
+                wsdlOperation.getOperationInput().getHeaders().stream().map(WSDLHeader::getPart).toList();
+        List<WSDLPart> inputBodyParts = wsdlOperation.getOperationInput().getMessage().getParts();
+        inputParts.addAll(inputHeaderParts);
+        inputParts.addAll(inputBodyParts);
+        for (WSDLPart inputPart : inputParts) {
+            List<Field> inputFields = partProcessor.generateFields(inputPart);
+            fields.addAll(inputFields);
+        }
+        fields.add(processOperationInput(wsdlOperation, fields));
+        return fields;
+    }
+
+    List<Field> generateOutputFields() {
+        List<String> processedOpOutputNames =
+                processedOperations.stream().map(op -> op.getOperationOutput().getName()).toList();
+        if (processedOpOutputNames.contains(wsdlOperation.getOperationOutput().getName())) {
+            return new ArrayList<>();
+        }
+        List<Field> fields = new ArrayList<>();
+        PartProcessor partProcessor = new PartProcessor();
+
+        List<WSDLPart> outputParts = new ArrayList<>();
+        List<WSDLPart> outputHeaderParts =
+                wsdlOperation.getOperationInput().getHeaders().stream().map(WSDLHeader::getPart).toList();
+        List<WSDLPart> outputBodyParts = wsdlOperation.getOperationOutput().getMessage().getParts();
+        outputParts.addAll(outputHeaderParts);
+        outputParts.addAll(outputBodyParts);
+        for (WSDLPart outputPart : outputParts) {
+            List<Field> outputFields = partProcessor.generateFields(outputPart);
+            fields.addAll(outputFields);
+        }
+        fields.add(processOperationOutput(wsdlOperation, fields));
+        return fields;
+    }
+
+    private ComplexField processOperationInput(WSDLOperation operation, List<Field> processedFields) {
+        XMLNSAttribute soapAttribute = new XMLNSAttribute(soapVersion == SOAPVersion.SOAP11 ?
+                "http://schemas.xmlsoap.org/soap/envelope/" : "http://www.w3.org/2003/05/soap-envelope");
+        soapAttribute.setPrefix("soap");
```

Comment:
Shall we move these `soap`, `Header`, and `Body` words into Const and reuse those?